File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        priority_queue<pair<int, vector<int>>, vector<pair<int, vector<int>>>, greater<pair<int, vector<int>>>> pq;

        vector<vector<int>> ans;
        for(vector<int> temp : points){
            int d =0;
            d = pow(temp[0],2)+pow(temp[1],2);
            pq.push({d,{temp[0],temp[1]}});
        }
        while(!pq.empty()){
           auto t = pq.top();
           pq.pop();
           ans.push_back(t.second);
           if(ans.size()==k) return ans;
        }
        return ans;
        
    }
};

File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        PriorityQueue<int[]> pq = new PriorityQueue<>((a,b)->Integer.compare(b[2], a[2]));
        for(int[] point: points){
            pq.add(new int[]{point[0],point[1],(point[0]*point[0])+(point[1]*point[1])});
            if(pq.size()>k) pq.poll();
        }
        int[][] ans = new int[pq.size()][2];
        int i=0;
        while(!pq.isEmpty())
        {
            int[] temp = pq.poll();
            ans[i][0]=temp[0];
            ans[i][1]=temp[1];
            i++;

        }
        return ans;
        
    }
}